infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -316,6 +316,34 @@ public synchronized void resetServer(CorfuPayloadMsg<Long> msg, ChannelHandlerCo
     }
 
     /**
+     * Consider a hint given by a runtime about the loction of the stream tails
+     *
+     * If the stream tail hinted is lower that the current one, the hint will be discarded.
+     *
+     * @param msg
+     * @param ctx
+     * @param r
+     * @param isMetricsEnabled
+     */
+    @ServerHandler(type=CorfuMsgType.TAILS_HINT, opTimer=metricsPrefix + "tails-hint")
```

Comment:
please change this to a "conservative" approach: send the global log tail along with the hint. If the current global tail moved beyond it, ignore this hint.

I am concerned about a slow hint that arrives after more entries are added to a stream, but the sequencer fails. Example:

stream has entries 1, 2
system crashes
on recovery, client discovers stream tail is 2 , but goes to sleep
entry 3 is added
sequencer fails and recovers
client sends hint 2 to new sequencer
entry 4 is added with backpointer to 2